daemon/endpoint.go

```diff
@@ -326,10 +327,13 @@ func (h *patchEndpointID) Handle(params PatchEndpointIDParams) middleware.Respon
 		if ep.GetStateLocked() == endpoint.StateReady {
 			ep.SetStateLocked(endpoint.StateWaitingToRegenerate, "Forcing endpoint regeneration because identity is known while handling API PATCH")
 		}
+		if ep.GetStateLocked() == endpoint.StateWaitingToRegenerate {
```

Comment:
Maybe do
```
wait = ep.SetStateLocked(endpoint.StateWaitingToRegenerate, "Forcing endpoint regeneration because identity is known while handling API PATCH")
```

instead?

Reply:
Given that this function locks, then does a bunch of checks which may leave the endpoint in `waitingToRegenerate` state, then unlocks, I thought it possible for this to happen in one request, then shortly thereafter, a second PATCH request comes which finds the endpoint already in that state, in which case we can still legitimately wait on the original regeneration.